comp/collector/collector/collectorimpl/collector.go

```diff
@@ -132,6 +142,35 @@ func newCollector(deps dependencies) *collectorImpl {
 	return c
 }
 
+func getFirstSuffix(s string) string {
+	return filepath.Ext(strings.TrimSuffix(s, filepath.Ext(s)))
+}
+
+func (c *collectorImpl) fillFlare(fb flaretypes.FlareBuilder) error {
```

Comment:
I don't think this belongs here, collector isn't reading these files, autoconfig does. Would it make sense to move this to the autoconfig package instead?